### PR TITLE
Add percentage input, widget and validation.

### PR DESCRIPTION
**Pull request by samiwel**
### What is the context of this PR?
Implement a Percentage input field.

### How to review 
I have included a test_percentage schema that can be used to test the new input field.
The percentage field should accept integer input and be limited to 0-100 inclusive.
The percentage sign should appear after the input field.

### CODE REVIEW

File: app/schema/widgets/percentage_widget.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+class PercentageWidget(Widget):
+    def render(self, state):
+        widget_params = {
+            'answer': {
+                'name': self.name,
+                'id': self.id,
+                'label': state.schema_item.label,
+                'value': state.value if state.value is not None else '',
+                'placeholder': '',
```
Comment (benjefferies) [inline app/schema/widgets/percentage_widget.py]: Can we delete placeholder if it's not used

File: app/schema/widgets/percentage_widget.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+            },
+            'debug': {
```
Comment (benjefferies) [inline app/schema/widgets/percentage_widget.py]: We can delete the debug stuff as we can use the flask debug toolbar for this

File: app/templates/partials/widgets/percentage_widget.html [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+<div class="input-type input-type--percentage">
+
+    {% set input = {
+    'type': 'number',
+    'classes': 'input-type__input',
+    'hasType': true,
+    'value': answer.value|e,
+    'name': answer.name,
+    'id': answer.id,
+    'placeholder': answer.placeholder,
```
Comment (benjefferies) [inline app/templates/partials/widgets/percentage_widget.html]: Placeholder can be removed if it's not used

**Review by benjefferies — CHANGES_REQUESTED** [review]

**Comment by jonnyshaw89** [comment]
@samiwel Does the % also need to be displayed on the summary screen?

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]
All looks good other than the few minor points ben has pointed out.

**Review by jonnyshaw89 — APPROVED** [review]

**Review by bennpearson — APPROVED** [review]

File: app/templates/partials/widgets/relationship_widget.html [hunk 4]
```diff
@@ -25,7 +25,8 @@
               "value": option.value,
               "name": answer.name,
               "id": option_id,
-              "checked": "checked" if option.selected
+              "checked": "checked" if option.selected,
+              "pattern": "[0-9]*"
```
Comment (benjefferies) [inline app/templates/partials/widgets/relationship_widget.html]: Is this supposed to be here?

**Review by benjefferies — CHANGES_REQUESTED** [review]

File: app/templates/partials/widgets/relationship_widget.html (hunk 4 above)
Comment (hamishtaplin) [reply app/templates/partials/widgets/relationship_widget.html]: Err, no.

Comment (samiwel) [reply app/templates/partials/widgets/relationship_widget.html]: Not sure, @hamishtaplin should the relationship widget be changed as well? can you confirm if this is correct?

Comment (samiwel) [reply app/templates/partials/widgets/relationship_widget.html]: Cool, I'll revert the change in this file

**Review by benjefferies — APPROVED** [review]

**Comment by benjefferies** [comment]
@samiwel can you squash?

**Comment by samiwel** [comment]
@benjefferies squashed!

**Pull request merged**